### PR TITLE
[10.0] [ADD] hr_payroll_account_operating_unit

### PR DESCRIPTION
**Pull request by serpentcs-dev1**
HR Payroll Account with Operating Units
=======================================

This module introduces the following features:

* Adds Operating Unit (OU) to the account moves and its lines created by the payslip, based on the Operating Unit (OU) defined in the Employee's Contract.

### CODE REVIEW

**Comment by nikul-serpentcs** [comment]
@aheficent Could you please review code.

File: hr_payroll_account_operating_unit/models/hr_payslip.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                                   slip.contract_id.operating_unit_id.id})
+        return res
+
+    @api.multi
+    def action_payslip_done(self):
+        OU = None
+        for slip in self:
+            # Check that all slips are related to contracts
+            # that belong to the same OU.
+            if OU:
```
Comment (AaronHForgeFlow) [inline hr_payroll_account_operating_unit/models/hr_payslip.py]: Why don't compare directly?

File: hr_payroll_account_operating_unit/models/hr_payslip.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+
+class HrPayslip(models.Model):
+
+    _inherit = 'hr.payslip'
+
+    @api.multi
+    def write(self, vals):
+        res = super(HrPayslip, self).write(vals)
+        if 'move_id' in vals and vals['move_id']:
```
Comment (AaronHForgeFlow) [inline hr_payroll_account_operating_unit/models/hr_payslip.py]: you can use `vals.get('move_id', False)` but it's the same

File: hr_payroll_account_operating_unit/models/hr_payslip.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    @api.multi
+    def action_payslip_done(self):
+        OU = None
+        for slip in self:
+            # Check that all slips are related to contracts
+            # that belong to the same OU.
+            if OU:
+                if slip.contract_id.operating_unit_id.id != OU:
+                    raise UserError(_('Configuration error!\nThe Contracts must\
```
Comment (AaronHForgeFlow) [inline hr_payroll_account_operating_unit/models/hr_payslip.py]: The \n is not needed just close the quotes and continue in the next line

File: hr_payroll_account_operating_unit/models/hr_payslip.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    def write(self, vals):
+        res = super(HrPayslip, self).write(vals)
+        if 'move_id' in vals and vals['move_id']:
+            for slip in self:
+                if slip.contract_id and slip.contract_id.operating_unit_id:
+                    slip.move_id.write({'operating_unit_id':
+                                        slip.contract_id.operating_unit_id.id})
+                    if slip.move_id.line_ids:
+                        slip.move_id.line_ids.\
+                            write({'operating_unit_id':
```
Comment (AaronHForgeFlow) [inline hr_payroll_account_operating_unit/models/hr_payslip.py:27]: Won't be better to include this in the account_move class?

**Review by AaronHForgeFlow — COMMENTED** [review]
Only code review :neutral_face:

File: hr_payroll_account_operating_unit/models/hr_payslip.py (hunk 4 above)
Comment (nikul-serpentcs) [reply hr_payroll_account_operating_unit/models/hr_payslip.py:27]: I didn't get you?! Do you want me to move this whole thing to while account_move are created?

File: hr_payroll_account_operating_unit/models/hr_payslip.py (hunk 1 above)
Comment (nikul-serpentcs) [reply hr_payroll_account_operating_unit/models/hr_payslip.py]: This is comparing the OU in contracts between the Payslip. I don't see the use case for this though.

Btw, what changes do you want me to make?

File: hr_payroll_account_operating_unit/models/hr_payslip.py (hunk 4 above)
Comment (AaronHForgeFlow) [reply hr_payroll_account_operating_unit/models/hr_payslip.py:27]: In v9 it was done by passing the operating unit in the context and then overriding the create method in the account_move. But maybe this is good so leave it like this.

Comment (nikul-serpentcs) [reply hr_payroll_account_operating_unit/models/hr_payslip.py:27]: Ok, Thanks

File: hr_payroll_account_operating_unit/models/hr_payslip.py (hunk 1 above)
Comment (AaronHForgeFlow) [reply hr_payroll_account_operating_unit/models/hr_payslip.py]: I was wondering if there is a way to map the contracts by different OU and if there was more than one raise the error. But I'm not really sure and this works. It's ok to me.

**Comment by AaronHForgeFlow** [comment]
Code looks now. We have to wait to the dependencies are merged.

**Comment by AaronHForgeFlow** [comment]
Superseded by #127

**Comment by mreficent** [comment]
Can be closed. Superseeded by https://github.com/OCA/operating-unit/pull/127.

**Pull request closed without merging**